components/HeroBlockTechnology/HeroBlockTechnology.tsx

```diff
@@ -24,15 +40,52 @@ function HeroBlockTechnology({
           <h3 className="hero-block-technology__title">{title}</h3>
           <div className="hero-block-technology__description">{description}</div>
 
-          <PrimaryButton className={`hero-block-technology__button hero-block-technology__button--${slicePathname}`}>
+          <PrimaryButton onClick={() => setIsOpen(true)} className={`hero-block-technology__button hero-block-technology__button--${slicePathname}`}>
             Оставить заявку
           </PrimaryButton>
+          {isOpen && (
+            <Modal
+              maxWidth="1008px"
+              title={!isSubmit ? t('modalTitle') : t('modalTitleSuccessful')}
+              subtitle={!isSubmit
+                ? t('modalSubtitle')
+                : `${t('modalSubtitleSuccessfulFirst')} ${email} ${t('modalSubtitleSuccessfulSecond')}`}
+              subtitleClassName="modal__subtitle--technology"
+              content={(
+                <>
+                  {!isSubmit && (
+                    <Form onSubmit={onFormSubmit} buttonClassName={`hero-block-technology__button hero-block-technology__button--${slicePathname}`} />
+                  )}
+                  {isSubmit && (
+                    <div className="discussion__list">
+                      <div className="title-type-4 discussion__list-title">{t('modalListTitle')}</div>
+                      <List steps={dataList} />
+                    </div>
+                  )}
+                </>
+              )}
```

Comment:
It seems that the BEM methodology is being violated here. 
I think that this div can be put into a separate component, and then it will solve the problems with classes